flycheck.el

```diff
@@ -6068,6 +6069,33 @@ See URL `http://www.foodcritic.io'."
        ;; http://matschaffer.github.io/knife-solo/#label-Init+command
        (locate-dominating-file parent-dir "cookbooks")))))
 
+(flycheck-define-checker chicken-scheme
+  "A CHICKEN Scheme syntax checker using the CHICKEN compiler `csc'.
+
+See URL `http://call-cc.org/'."
+  :command ("csc" "-A" "-local" source)
+  :predicate
+  (lambda ()
+    (or (not (eq major-mode 'scheme-mode))
+        ;; In `scheme-mode' we must check the current Scheme implementation
+        ;; being used
+        (and (boundp 'geiser-impl--implementation)
+             (eq geiser-impl--implementation 'chicken))))
```

Comment:
Is there a reason to keep the `(not (eq major-mode 'scheme-mode))` form? I'm also not sure why the Racket checker has that form instead of `(eq major-mode 'racket-mode)`.